pkg/auth/handler/userverify/verify_request.go

```diff
@@ -56,16 +59,32 @@ func (f VerifyRequestHandlerFactory) ProvideAuthzPolicy() authz.Policy {
 	)
 }
 
-type VerifyRequestPayload struct {
-	LoginIDKey string `json:"login_id_key"`
-	LoginID    string `json:"login_id"`
+type loginIDType string
+
+const (
+	loginIDTypeEmail loginIDType = "email"
+)
+
+var allLoginIDTypes = []string{
+	string(loginIDTypeEmail),
 }
 
-func (payload VerifyRequestPayload) Validate() error {
-	if payload.LoginIDKey == "" {
-		return skyerr.NewInvalidArgument("empty login ID key", []string{"login_id_key"})
+func (t loginIDType) MatchPrincipal(principal *password.Principal, provider password.Provider) bool {
+	switch t {
+	case loginIDTypeEmail:
+		return provider.CheckLoginIDKeyType(principal.LoginIDKey, metadata.Email)
+	default:
+		return false
 	}
+}
+
+type VerifyRequestPayload struct {
+	RawLoginIDType string      `json:"login_id_type"`
+	LoginIDType    loginIDType `json:"-"`
```

Comment:
`RawLoginIDType` seems to redundant. We have `Validate` so we can validate `LoginIDType loginIDType` before `Handle` can ever be run